pseudomonas_analysis/nbconverted/explore_expression_data.py

```diff
@@ -0,0 +1,634 @@
+# 
+# A similar trend is seen for the volcano plot that is completely flat as well. There is more noise in the gene expression data.
+
+# ## Expression of template experiment
+# 
+# Do the samples have a clear separation in gene space (WT vs mutant)?
+
+# In[27]:
+
+
+normalized_compendium_data = pd.read_csv(normalized_compendium_filename, sep="\t", index_col=0, header=0)
+template_data = pd.read_csv(template_filename, sep="\t", index_col=0, header=0)
+
+
+# In[28]:
+
+
+print(template_data.shape)
+template_data
+
+
+# In[29]:
+
+
+# If template experiment included in training compendium
+# Get normalized template data
+sample_ids = list(template_data.index)
+normalized_template_data = normalized_compendium_data.loc[sample_ids]
+
+print(normalized_template_data.shape)
+normalized_template_data.head()
+
+
+# In[30]:
+
+
+# Label samples 
+wt_sample_ids = ["GSM822708_wtLB_A.CEL", "GSM822709_wtLB_B.CEL"]
+mutant_sample_ids = ["GSM822712_delta_cbrBLB_A.CEL", "GSM822713_delta_cbrBLB_B.CEL"]
+normalized_compendium_data['sample group'] = "compendium"
+normalized_template_data.loc[wt_sample_ids, 'sample group'] = "template_WT"
+normalized_template_data.loc[mutant_sample_ids, 'sample group'] = "template_mutant"
+
+
+# In[31]:
+
+
+normalized_all_data = pd.concat([normalized_template_data,
+                                 normalized_compendium_data
+])
+
+
+# In[ ]:
+
+
+# Plot
+
+# Drop label column
+normalized_all_data_numeric = normalized_all_data.drop(['sample group'], axis=1)
+
+model = umap.UMAP(random_state=1).fit(normalized_all_data_numeric)
+
+normalized_all_data_UMAPencoded = model.transform(normalized_all_data_numeric)
+normalized_all_data_UMAPencoded_df = pd.DataFrame(data=normalized_all_data_UMAPencoded,
+                                         index=normalized_all_data.index,
+                                         columns=['1','2'])
+
+# Add back label column
+normalized_all_data_UMAPencoded_df['sample group'] = normalized_all_data['sample group']
+
+# Plot
+fig = pn.ggplot(normalized_all_data_UMAPencoded_df, pn.aes(x='1', y='2'))
+fig += pn.geom_point(pn.aes(color='sample group'), alpha=0.4)
+fig += pn.labs(x ='UMAP 1',
+            y = 'UMAP 2',
+            title = 'Gene expression data in gene space')
+fig += pn.theme_bw()
+fig += pn.theme(
+    legend_title_align = "center",
+    plot_background=pn.element_rect(fill='white'),
+    legend_key=pn.element_rect(fill='white', colour='white'), 
+    legend_title=pn.element_text(family='sans-serif', size=15),
+    legend_text=pn.element_text(family='sans-serif', size=12),
+    plot_title=pn.element_text(family='sans-serif', size=15),
+    axis_text=pn.element_text(family='sans-serif', size=12),
+    axis_title=pn.element_text(family='sans-serif', size=15)
+    )
+fig += pn.scale_color_manual(['#bdbdbd', 'red', 'blue'])
+fig += pn.guides(colour=pn.guide_legend(override_aes={'alpha': 1}))
+
+fig += pn.scales.xlim(9,10)
+print(fig)
```

Comment:
I can't see the plot here (but I'll take your word for what you've written)

Reply:
I'll re-run it so its there. This was in the previous PR you reviewed and didn't change.